src/gwt/acesupport/acemode/utils.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * utils.js
+ *
+ * Copyright (C) 2009-12 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+define("mode/utils", function(require, exports, module) {
+
+(function() {
+
+   var that = this;
+
+   this.contains = function(array, object)
+   {
+      for (var i = 0; i < array.length; i++)
+         if (array[i] === object)
+            return true;
+
+      return false;
+   };
+
+   this.isArray = function(object)
+   {
+      return Object.prototype.toString.call(object) === '[object Array]';
+   };
+
+   this.getPrimaryState = function(session, row, state)
```

Comment:
It looks like we never use the 3rd argument (neither implementation nor callers). Should we remove it?